Yoast/yoast-components
# Add prop to disable and hide marking buttons

## Summary

This PR can be summarized in the following changelog entry:

* Added a prop to the `ContentAnalysis` to disable or hide the eye buttons.

## Relevant technical choices:

* I've reset the cursor to `default` rather than `pointer` on hovering a disabled button. This is consistent with what WP is doing. 
* I've changed the icon color of the disabled button to #ddd ($color_grey) instead of #e6e6e6 (which is not in the colors.json). #ddd is a little darker, but the background is darker as well,
 because now there is a button background instead of the white meta box background. The color contrast is 1.27 instead of 1.25 🎉 .
* The design of the disabled button has been approved by @hedgefield.


## Test instructions

This PR can be tested by following these steps:

* Check out this branch.
* `yarn install`,`yarn start`
* Go to `localhost:3333`
* Change the marksButtonStatus in `app/ContentAnalysisWrapper.js` to 'hidden' and 'disabled' and check whether the buttons are hidden or disabled.

Part of the fix for https://github.com/Yoast/wordpress-seo/issues/8442
Fixes #408
